mis_funciones.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+* @file mis_funciones.cpp
+* @version 1.0
+* @date 18/7/24
+* @author Ramos Leonardo
+*/
+#include "mis_funciones.hpp"
+
+/// @cond VARIABLES EXTERNAS
+extern BusOut keypadRows;                         
+extern BusIn keypadCols;                         
+extern DigitalOut workingLed;                     
+extern DigitalOut accessGrantedLed;              
+extern DigitalOut accessDeniedLed;               
+extern UnbufferedSerial uartUSB;                 
+
+extern bool sequenceDetectedState;                      
+extern bool accessState;                                
+extern bool timeoutState;                               
+extern int colReading;                                
+extern int rowReading;                                  
+extern uint8_t currentIndex;                            
+extern uint16_t accumulatedTime;                        
+extern char inputSequence[SEQUENCE_LEN + 1];            
+extern char keypadAllowedEntries[ROW_LEN][COL_LEN];    
+/// @endcond
+
+/**
+ * @brief keypadInit Inicializa el keypad 4x4 para su uso. Principalmente
+ *        configura los GPIO de columnas como input PULL DOWN y da valores iniciales
+ *        a ciertos flags. 
+ */
+void keypadInit() {
+  keypadCols.mode(PullDown);
+  inputSequence[SEQUENCE_LEN] = '\0';
+  sequenceDetectedState = false;
+  currentIndex = 0;
+  accumulatedTime = 0;
+}
+
+/**
+ * @brief uartInit Configura velocidad de transmisión y formato de la trama
+ *        de datos de UART
+ */
+void uartInit() {
+  uartUSB.baud(9600);
+  uartUSB.format(/*bits*/ 8,
+                 /*paridad*/ SerialBase::None,
+                 /*bits de stop*/ 1);
+}
+
+/**
+ * @brief keypadSweepUpdate Barre el keypad 4x4 hasta encontrar
+ *        una tecla presionada, guarda el caracter asociado
+ *        a esa tecla e indica mediante flag sequenceDetectedState
+ *        que se ha ingresado la secuencia completa
+ */
+void keypadSweepUpdate() {
+  for (uint8_t i = 0; i < ROW_LEN; i++) {
+    keypadRows = (1 << i);
+    if (keypadCols != 0) {
+      delay(100);
+      if (keypadCols != 0) {
+        timeoutState = true;
+        accumulatedTime = 0;
+        rowReading = keypadRows.read();
+        colReading = keypadCols.read();
+        inputSequence[currentIndex] = getKeyPressed();
+        currentIndex++;
+        if (currentIndex >= SEQUENCE_LEN) {
+          uartUSB.write("\n", 1);  
+          sequenceDetectedState = true;
+        }
+        i = ROW_LEN;
```

Comment:
Es muy mala práctica tocar la variable del `for`
Hay que leer. Kernighan y Ritchie se van a deprimir si no los leen.
Si querés salir del for, existe `break`

Reply:
Conozco la existencia del break. Esta práctica la acarreo de muchos profesores que consideraban que un bucle debe terminar por su condición, y no de forma abrupta (de hecho hay muchas discusiones en foros sobre esto). Gracias por el comentario igualmente, siempre hay que adecuarse al estándar que maneja un equipo.